src/test/java/edu/tamu/eider/ApiDocumentation.java

```diff
@@ -0,0 +1,37 @@
+package edu.tamu.eider;
+
+import static org.springframework.restdocs.mockmvc.MockMvcRestDocumentation.documentationConfiguration;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.restdocs.RestDocumentationContextProvider;
+import org.springframework.restdocs.RestDocumentationExtension;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@SpringBootTest
+@ExtendWith({ RestDocumentationExtension.class, SpringExtension.class })
+public class ApiDocumentation {
+
+    @Autowired
+    protected WebApplicationContext context;
+
+    protected ObjectMapper mapper;
+
+    protected MockMvc mockMvc;
+
+    @BeforeEach
+    public void setUp(RestDocumentationContextProvider restDocumentation) {
+        this.mapper = new ObjectMapper();
+        this.mockMvc = MockMvcBuilders.webAppContextSetup(context)
```

Comment:
RestDocs can be autoconfigured.

https://docs.spring.io/spring-boot/docs/current/reference/html/spring-boot-features.html#boot-features-testing-spring-boot-applications-testing-with-mock-environment

Reply:
This can be accomplished with full app context with the following annotations.
```
@SpringBootTest
@AutoConfigureMockMvc
@AutoConfigureRestDocs
```

If desired to isolate the test to a specific controller `@SpringBootTest` annotation can be replaced with `@WebMvcTest(EntityController.class)`. Any beans EntityController autowires would require mocking. To many tests running entire application context makes for slow tests and bloated test logs.

Reply:
It appears the manual build MockMvc is excluding the Spring Security Filter Chain allowing the tests to not require authentication. I think the generated Rest Docs will be unaware of the basic auth requirement of these enpoints.

If still wanting to manually build MockMvc the security chain can be added like:

```
@Autowired
private FilterChainProxy springSecurityFilterChain;
```
and before building mockMvc
`.apply(springSecurity(springSecurityFilterChain))`

Reply:
The tests will have to be updated to include http basic credentials. Such as:

`.with(httpBasic("admin", "admin")))`